### PR TITLE
Add opt-in analytics to onboarding tasks

### PR DESCRIPTION
**Pull request by justjanne**
Type: Enhancement
Closes: https://github.com/vector-im/element-web/issues/21705

## In Short
* opt-in via our regular opt-in flow for analytics
* adds interaction event tracking to “create new DM”, “create new Room” and “explore Rooms” dialogs
* allows distinguishing whether these events happened from the onboarding buttons, via the room list, or via other sources

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add opt-in analytics to onboarding tasks ([\#8409](https://github.com/matrix-org/matrix-react-sdk/pull/8409)). Fixes vector-im/element-web#21705. Contributed by @justjanne.

Preview: https://pr8409--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/structures/LeftPanel.tsx [hunk 1]
```diff
@@ -116,8 +118,9 @@ export default class LeftPanel extends React.Component<IProps, IState> {
         dis.fire(Action.OpenDialPad);
     };
 
-    private onExplore = () => {
+    private onExplore = (ev: ButtonEvent) => {
         dis.fire(Action.ViewRoomDirectory);
+        PosthogTrackers.trackInteraction("WebLeftPanelExploreRoomsItem", ev);
```
Comment (t3chguy) [inline src/components/structures/LeftPanel.tsx:123]: Hmm `...Item` is normally used for context menu items, not for buttons which are visible at the top level

Comment (justjanne) [reply src/components/structures/LeftPanel.tsx:123]: That’s a good point, though we probably should’ve caught that mistake in https://github.com/matrix-org/matrix-analytics-events/pull/54 ?

Comment (t3chguy) [reply src/components/structures/LeftPanel.tsx:123]: I think it isn't up to the team reviewing the Analytics events repo for checking where the app has a given button, especially when the partner PR doesn't exist to see where you'll hook it up.

**Review by t3chguy — APPROVED** [review]
You will likely need to update the matrix-analytics-events hash in package.json & yarn.lock due to how yarn handles github dependencies

**Pull request merged**